plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSink.java

```diff
@@ -117,11 +118,11 @@ public IcebergPageSink(
         requireNonNull(inputColumns, "inputColumns is null");
         this.outputSchema = requireNonNull(outputSchema, "outputSchema is null");
         this.partitionSpec = requireNonNull(partitionSpec, "partitionSpec is null");
-        this.outputPath = requireNonNull(outputPath, "outputPath is null");
+        this.locationProvider = requireNonNull(locationProvider, "locationProvider is null");
         this.fileWriterFactory = requireNonNull(fileWriterFactory, "fileWriterFactory is null");
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.hdfsContext = requireNonNull(hdfsContext, "hdfsContext is null");
-        this.jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(outputPath)));
+        this.jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(locationProvider.newDataLocation(""))));
```

Comment:
I'd prefer continuing to pass `outputPath` in to the cnstr just to be used here. `locationProvider.newDataLocation("")` being the parent directory is not guaranteed to be true.

Reply:
The reason this is passed in instead of the `outputPath` is because the `outputPath` is computed as `table.location() + "/data"`, but in ObjectStorageLocationProvider this might be wrong, the data might sit in another S3 bucket, and the security mapping will change when this `hdfsEnvironment.getConfiguration` is called:

https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingConfigurationProvider.java#L90-L100

For the default case, this I think is exactly the same path as the original `outputPath`:

https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/LocationProviders.java#L77

Reply:
> the data might sit in another S3 bucket

Ah, ok. I didn't think about that.